demo/demo.js

```diff
@@ -1,117 +1,64 @@
-
-    $categories.click(function() {
-      var $category = $(this);
-
-      $categories.removeClass('current');
-      $category.addClass('current');
-      selectCategory($category.data('category'));
-    });
-
-    $categories.first().click();
-  }
-
-  function selectStyle(style) {
-    $linkTitle.each(function(i, link) {
-      link.disabled = (link.title !== style);
-    });
-  }
-
-  function initStyles() {
-    var $styles;
+if (Worker && location.protocol !== "file:") {
+  const worker = new Worker("worker.js");
+  worker.addEventListener("message", (event) => {
+    const { language, html } = event.data;
+    document.querySelector(`.language-${language}`).innerHTML = html;
+  });
 
-    $linkTitle.each(function(i, link) {
-      $styleContainer.append('<li>' + link.title + '</li>');
+  highlight = (sample) => {
+    worker.postMessage({
+      language: sample.className.replace("hljs language-", ""),
+      code: sample.textContent,
     });
+  };
+} else {
+  highlight = (sample) => {
+    const language = sample.className.replace("hljs language-", "");
+    const { value } = hljs.highlight(language, sample.textContent);
+    document.querySelector(`.language-${language}`).innerHTML = value;
+  };
+}
+
+// Highlight all visible code samples first
+document.querySelectorAll("div:not(.hidden) .hljs").forEach(highlight);
+
+// Highlight all hidden code samples later
+setTimeout(() => document.querySelectorAll(".hidden .hljs").forEach(highlight), 0);
+
+document.querySelectorAll(".categories > li").forEach((category) => {
+  category.addEventListener("click", (event) => {
+    const current = document.querySelector(".categories .current");
+    const currentCategory = current.dataset.category;
+    const nextCategory = event.target.dataset.category;
+
+    if (currentCategory !== nextCategory) {
+      current.classList.remove("current");
+      event.target.classList.add("current");
+
+      document
+        .querySelectorAll(`.${currentCategory}`)
+        .forEach((language) => language.classList.add("hidden"));
+      document
+        .querySelectorAll(`.${nextCategory}`)
+        .forEach((language) => language.classList.remove("hidden"));
+    }
+  });
+});
 
-    $styles = $styleContainer.find('li');
-
-    $styles.click(function() {
-      var $style = $(this);
+document.querySelectorAll(".styles > li").forEach((style) => {
+  style.addEventListener("click", (event) => {
+    const current = document.querySelector(".styles .current");
+    const currentStyle = current.textContent;
+    const nextStyle = event.target.textContent;
 
-      $styles.removeClass('current');
-      $style.addClass('current');
-      selectStyle($style.text());
-    });
-    $styles.first().click();
-  }
+    if (currentStyle !== nextStyle) {
+      document.querySelector(`link[title="${nextStyle}"]`).removeAttribute("disabled");
+      document.querySelector(`link[title="${currentStyle}"]`).setAttribute("disabled", "disabled");
```

Comment:
Would `alternative` be better than disabled?  I don't really know but it's what I've used for this kind of thing in the past.

Reply:
Don't we still have to toggle `alternative` on/off?

Reply:
I can have a look, but here I just followed what the previous script did:
https://github.com/highlightjs/highlight.js/blob/7e108baaa3be0a7ad402d60259d060fcefa44021/demo/demo.js#L86-L90

Reply:
The current usage (or `alternative`/`disabled`) works fine in Firefox and Chromium. If you are on Mac, could you try it for Safari?

Reply:
Ah, I was looking more closely at the new code than the old... 

From MDN (learning something new):

> Alternate (rel="alternate stylesheet", title="..." must be specified): disabled by default, can be selected.

So alternates are simply disabled by default... I've always toggled their "alternate" status (making them active by default), but seems you can just as easily toggle their "disabled" status, thereby also making them active. :-)

So this seems a fine enough strategy.